### ISSUE TITLE
Android App Won't Launch / Freezes When CalDAV Server Unavailable

### ISSUE THREAD

**Issue by mwaterbu** [johannesjo/super-productivity#3057]
### Your Environment

- Version used: 21.0 apk (from Play Store)
- Operating System and version: Android 12

### Expected Behavior

App launches and is usable

### Current Behavior

App will sit on loading screen for several minutes. It may eventually show the task list but is unresponsive and shows an error at the bottom of the screen:
> Unknown Sync Error: Error: Invalid response: -3
Network Error

### Steps to Reproduce (for bugs)

1. Setup a CalDav server on local network
2. Connect one or more projects to this server
3. Disconnect from local network, but retain internet connectivity (don't go offline)
4. Attempt to launch app

### Can you reproduce this reliably?

This can be reliably reproduced. When offline completely, the app launches without issue. However, if there is a network connection where the CalDAV server is not available (example listed above), this bug will occur.

**Comment by mwaterbu** [comment]
To close the loop on this: I found that this was actually caused by the WebDAV backup sync, and only happened when I was on cellular data, not WiFi (even networks other than my home WiFi). As a workaround, I found that if I set the server to the local domain (hostname) instead of the local IP address, the issue no longer occurred. So I think my carrier was just ignoring the connection requests instead of returning an error. Ideally, it wouldn't even try to connect unless I was on my home network, but this would require a new setting and likely some app permissions.
